core/src/main/antlr4/com/databricks/labs/remorph/parsers/snowflake/SnowflakeParser.g4

```diff
@@ -151,6 +153,9 @@ copyIntoTable
     ) R_PAREN files? pattern? fileFormat? copyOptions*
     ;
 
+declare: DECLARE (id dataType (DEFAULT expr | (L_PAREN selectStatement R_PAREN))? SEMI)+
+    ;
```

Comment:
Split the rule in smaller components to makes visitor implementation easier:
```suggestion
declareClause: DECLARE declarationStatement+
    ;
 
 declarationStatement
    : id dataType                                                                #declarationSimple
    | id dataType DEFAULT expr                                       #declarationDefault
    | id dataType L_PAREN selectStatement R_PAREN #declarationSelect
    ;
```

This is just a sketch (you may want to factorize the `id dataType` part out at some point) but it would make the visitor implementation more straightforward:

In the method that uses the content of `declareClause` (and other stuff) to build a `ir.CreateProcedure`, you'd just have to do something like `ctx.declareClause().declareStatement().asScala.map(_.accept(statementBuilder))`. 

In `SnowflakeStatementBuilder` you would just have to implement three small methods for each case of `declareStatement`, namely `visitDeclareSimple(ctx: DeclareSimpleContext)`, `visitDeclareDefault(ctx: DeclareDefaultContext)` and `visitDeclareSelect(ctx: DeclareSelectContext)`.